cl/opinion_page/views.py

```diff
@@ -1233,21 +1238,25 @@ async def block_item(request: HttpRequest) -> HttpResponse:
     if is_ajax(request) and user.is_superuser:
         obj_type = request.POST["type"]
         pk = request.POST["id"]
-        if obj_type == "docket":
-            # Block the docket
-            d = await aget_object_or_404(Docket, pk=pk)
-            d.blocked = True
-            d.date_blocked = now()
-            await d.asave()
-        elif obj_type == "cluster":
-            # Block the cluster and the docket
+
+        if obj_type not in ["docket", "cluster"]:
+            return HttpResponseBadRequest(
+                f"This view can not handle the provided type"
+            )
+
+        if obj_type == "cluster":
+            # Block the cluster
             cluster = await aget_object_or_404(OpinionCluster, pk=pk)
             cluster.blocked = True
             cluster.date_blocked = now()
             await cluster.asave(index=False)
-            cluster.docket.blocked = True
-            cluster.docket.date_blocked = now()
-            await cluster.docket.asave()
+
+        docket_pk = pk if obj_type == "docket" else cluster.docket_id
```

Comment:
It seems that the cluster can be referenced before it is assigned, so we could consider doing something like:

`cluster = None`

```suggestion
        docket_pk = pk if obj_type == "docket" else cluster.docket_id if cluster else None
```